ConsoleAppAzureServices/ConsoleAppAzureServices/Program.cs

```diff
@@ -1,47 +1,49 @@
-﻿using System;
+﻿using Azure;
+using Azure.Identity;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Azure.Security.KeyVault.Secrets;
+using System;
 using System.Threading.Tasks;
-using Microsoft.Azure.KeyVault;
-using Microsoft.Azure.Management.ResourceManager;
-using Microsoft.Azure.Services.AppAuthentication;
-using Microsoft.Rest;
 
 namespace ConsoleAppAzureServices
 {
     class Program
     {
         static void Main()
         {
-            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
+            InteractiveBrowserCredential credential = new InteractiveBrowserCredential();
 
-            GetSecretFromKeyVault(azureServiceTokenProvider).Wait();
+            AuthenticationRecord authRecord = credential.Authenticate();
 
-            GetResourceGroups(azureServiceTokenProvider).Wait();
+            GetSecretFromKeyVault(credential).Wait();
 
-            if (azureServiceTokenProvider.PrincipalUsed != null)
+            GetResourceGroups(credential).Wait();
+
+            if (credential != null)
```

Comment:
`credential` seems not null here.

Reply:
@archerzz - I have updated, please review.